src/js/smtpserver.c

```diff
@@ -121,41 +455,383 @@ int init_header_class(JSContext *cx, JSObject *global) {
+		return -1;
+	}
+
+	// Add messages property
+	switch(JS_TypeOfValue(cx, messages)) {
+		case JSTYPE_STRING:
+			// Create the messages array property
+			messages_arr = JS_NewArrayObject(cx, 0, NULL);
+
+			if (!messages_arr) {
+				return -1;
+			}
+
+			// Add message to messages array
+			if (!JS_SetElement(cx, messages_arr, 0, &messages)) {
+				return -1;
+			}
+
+			// Copy the messages to the property
+			jsval aux = OBJECT_TO_JSVAL(messages_arr);
+			if (!JS_SetProperty(cx, response_obj, "messages", &aux)) {
+				return -1;
+			}
+
+			break;
+		case JSTYPE_OBJECT:
+			// Copy the messages to the property
+			if (!JS_SetProperty(cx, response_obj, "messages", &messages)) {
+				return -1;
+			}
+			break;
+		default:
+			return JS_FALSE;
+	}
+
+	// Add disconnect property
+	if (!JS_SetProperty(cx, response_obj, "disconnect", &disconnect)) {
+		return -1;
+	}
+
+	response = OBJECT_TO_JSVAL(response_obj);
+
+	JS_SET_RVAL(cx, vp, response);
+	return JS_TRUE;
+}
+
+int init_smtp_response_class(JSContext *cx, JSObject *global) {
+	static JSClass smtp_response_class = {
+	    "SmtpResponse", 0,
+	    JS_PropertyStub, JS_PropertyStub, JS_PropertyStub, JS_PropertyStub,
+	    JS_EnumerateStub, JS_ResolveStub, JS_ConvertStub, JS_PropertyStub,
+	    NULL, NULL, NULL, response_construct, NULL, NULL, NULL, NULL
+	};
+
+	JSObject *smtpResponseClass;
+
+	// Create the SmtpPath class
+	smtpResponseClass = JS_InitClass(cx, global, NULL, &smtp_response_class, response_construct, 1, NULL, NULL, NULL, NULL);
+
+	if (!smtpResponseClass) {
+		return -1;
+	}
+
+	return 0;
+}
+
+static int connect_to_address(char *ip, char *port)
+{
+	int sockfd, status, portno;
+	struct sockaddr_in serv_addr;
+	struct hostent *server;
+
+	portno = atoi(port);
+	sockfd = socket(AF_INET, SOCK_STREAM, 0);
+
+	if (sockfd < 0) {
+		// throw exc
+		printf("sock failed\n");
+		return -1;
+	}
+
+	server = gethostbyname(ip);
+
+	if (!server) {
+		// throw exc
+		printf("server failed\n");
+	}
+
+	bzero((char*) &serv_addr, sizeof(serv_addr));
+
+	serv_addr.sin_family = AF_INET;
+
+	bcopy((char *) server->h_addr, (char *) &serv_addr.sin_addr.s_addr, server->h_length);
+	serv_addr.sin_port = htons(portno);
+
+	if (connect(sockfd, (struct sockaddr *) &serv_addr, sizeof(serv_addr)) < 0) {
+		// throw new error
+		printf("connect failed\n");
+	}
+
+	return sockfd;
+}
+
+static JSBool smtpClient_connect(JSContext *cx, unsigned argc, jsval *vp) {
+	jsval host, port, client, connection;
+	char *c_host, *c_port;
+	int sockfd;
+
+	client = JS_THIS(cx, vp);
+
+	// Get host
+	if (!JS_GetProperty(cx, JSVAL_TO_OBJECT(client), "host", &host)) {
+		return JS_FALSE;
+	}
+
+	// Get port
+	if (!JS_GetProperty(cx, JSVAL_TO_OBJECT(client), "port", &port)) {
+		return JS_FALSE;
+	}
+
+	c_host = JS_EncodeString(cx, JSVAL_TO_STRING(host));
+	c_port = JS_EncodeString(cx, JSVAL_TO_STRING(port));
+
+	sockfd = connect_to_address(c_host, c_port);
+	connection = INT_TO_JSVAL(sockfd);
```

Comment:
This code seems to be outdated, since you mentioned in your last email that you changed it to use bfd_t for the whole smtp_client API. Please push the bfd_t related changes, so I can review. Thanks!